File: app/src/main/java/com/example/geotaskaireminder/model/TaskEntity.java
package com.example.geotaskaireminder.model;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "tasks")
public class TaskEntity {
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "id")
    private long id;

    @ColumnInfo(name = "title")
    private String title = "";

    @ColumnInfo(name = "notes")
    private String notes = "";

    @ColumnInfo(name = "location_name")
    private String locationName = "";

    @ColumnInfo(name = "priority", defaultValue = "'Medium'")
    private String priority = "Medium";

    @ColumnInfo(name = "latitude")
    private double latitude;

    @ColumnInfo(name = "longitude")
    private double longitude;

    @ColumnInfo(name = "radius_meters")
    private double radiusMeters;

    @ColumnInfo(name = "created_at")
    private long createdAt;

    @ColumnInfo(name = "updated_at")
    private long updatedAt;

    @ColumnInfo(name = "reminder_triggered", defaultValue = "0")
    private boolean reminderTriggered;

    public TaskEntity() {
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getNotes() {
        return notes;
    }

    public void setNotes(String notes) {
        this.notes = notes;
    }

    public String getLocationName() {
        return locationName;
    }

    public void setLocationName(String locationName) {
        this.locationName = locationName;
    }

    public String getPriority() {
        return priority;
    }

    public void setPriority(String priority) {
        this.priority = priority;
    }

    public double getLatitude() {
        return latitude;
    }

    public void setLatitude(double latitude) {
        this.latitude = latitude;
    }

    public double getLongitude() {
        return longitude;
    }

    public void setLongitude(double longitude) {
        this.longitude = longitude;
    }

    public double getRadiusMeters() {
        return radiusMeters;
    }

    public void setRadiusMeters(double radiusMeters) {
        this.radiusMeters = radiusMeters;
    }

    public long getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(long createdAt) {
        this.createdAt = createdAt;
    }

    public long getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(long updatedAt) {
        this.updatedAt = updatedAt;
    }

    public boolean isReminderTriggered() {
        return reminderTriggered;
    }

    public void setReminderTriggered(boolean reminderTriggered) {
        this.reminderTriggered = reminderTriggered;
    }
}

File: app/src/main/java/com/dinesh/geotaskai/data/TaskEntity.kt
package com.dinesh.geotaskai.data

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "tasks")
data class TaskEntity(
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "id")
    var id: Long = 0L,

    @ColumnInfo(name = "title")
    var title: String? = "",

    @ColumnInfo(name = "notes")
    var notes: String? = "",

    @ColumnInfo(name = "location_name")
    var locationName: String? = "",

    @ColumnInfo(name = "priority", defaultValue = "'Medium'")
    var priority: String? = "Medium",

    @ColumnInfo(name = "latitude")
    var latitude: Double = 0.0,

    @ColumnInfo(name = "longitude")
    var longitude: Double = 0.0,

    @ColumnInfo(name = "radius_meters")
    var radiusMeters: Double = 0.0,

    @ColumnInfo(name = "created_at")
    var createdAt: Long = 0L,

    @ColumnInfo(name = "updated_at")
    var updatedAt: Long = 0L,

    @ColumnInfo(name = "reminder_triggered", defaultValue = "0")
    var reminderTriggered: Boolean = false,
)
